sw/applications/example_spi_flash_write_quad/main.c

```diff
@@ -0,0 +1,446 @@
+spi_host_t spi_host;
+
+// Test buffers
+uint32_t flash_data_towrite[16] = {
+    0x76543210,0xfedcba98,0x579a6f90,0x657d5bee,0x758ee41f,0x01234567,0xfedbca98,
+    0x89abcdef,0x679852fe,0xff8252bb,0x763b4521,0x6875adaa,0x09ac65bb,0x666ba334,
+    0x44556677,0x0000ba98
+};
+uint32_t flash_data_toread[16] = {};
+
+// Function to set the QE bit in the W25Q128JW flash memory
+void set_QE_bit(void);
+
+#ifndef USE_SPI_FLASH
+void fic_irq_spi(void)
+{
+    // Disable SPI interrupts
+    spi_enable_evt_intr(&spi_host, false);
+    spi_enable_rxwm_intr(&spi_host, false);
+    spi_intr_flag = 1;
+}
+#else
+void fic_irq_spi_flash(void)
+{
+    // Disable SPI interrupts
+    // PRINTF("&");
+    spi_enable_evt_intr(&spi_host, false);
+    spi_enable_rxwm_intr(&spi_host, false);
+    spi_intr_flag = 1;
+}
+#endif //USE_SPI_FLASH
+
+
+
+static inline __attribute__((always_inline)) void spi_config() {
+/*
+* SPI CONFIGURATIONS
+*/
+#ifndef USE_SPI_FLASH
+    spi_host.base_addr = mmio_region_from_addr((uintptr_t)SPI_HOST_START_ADDRESS);
+#else
+    spi_host.base_addr = mmio_region_from_addr((uintptr_t)SPI_FLASH_START_ADDRESS);
+#endif //USE_SPI_FLASH
+
+    soc_ctrl_t soc_ctrl;
+    soc_ctrl.base_addr = mmio_region_from_addr((uintptr_t)SOC_CTRL_START_ADDRESS);
+    uint32_t core_clk = soc_ctrl_get_frequency(&soc_ctrl);
+
+    // Enable interrupt on processor side
+    // Enable global interrupt for machine-level interrupts
+    CSR_SET_BITS(CSR_REG_MSTATUS, 0x8);
+    // Set mie.MEIE bit to one to enable machine-level fast spi interrupt
+
+
+#ifndef USE_SPI_FLASH
+        const uint32_t mask = 1 << 20;
+#else
+        const uint32_t mask = 1 << 21;
+#endif
+    CSR_SET_BITS(CSR_REG_MIE, mask);
+    spi_intr_flag = 0;
+
+#ifdef USE_SPI_FLASH
+        // Select SPI host as SPI output
+        soc_ctrl_select_spi_host(&soc_ctrl);
+#endif // USE_SPI_FLASH
+
+
+    CSR_SET_BITS(CSR_REG_MIE, mask);
+
+    // Enable SPI host device
+    spi_set_enable(&spi_host, true);
+    // Enable SPI output
+    spi_output_enable(&spi_host, true);
+
+    // Configure SPI clock
+    // SPI clk freq = 1/2 core clk freq when clk_div = 0
+    // SPI_CLK = CORE_CLK/(2 + 2 * CLK_DIV) <= CLK_MAX => CLK_DIV > (CORE_CLK/CLK_MAX - 2)/2
+    uint16_t clk_div = 0;
+    if(FLASH_CLK_MAX_HZ < core_clk/2){
+        clk_div = (core_clk/(FLASH_CLK_MAX_HZ) - 2)/2; // The value is truncated
+        if (core_clk/(2 + 2 * clk_div) > FLASH_CLK_MAX_HZ) clk_div += 1; // Adjust if the truncation was not 0
+    }
+    // SPI Configuration
+    // Configure chip 0 (flash memory)
+    const uint32_t chip_cfg = spi_create_configopts((spi_configopts_t){
+        .clkdiv     = clk_div,
+        .csnidle    = 0xF,
+        .csntrail   = 0xF,
+        .csnlead    = 0xF,
+        .fullcyc    = false,
+        .cpha       = 0,
+        .cpol       = 0
+    });
+    spi_set_configopts(&spi_host, 0, chip_cfg);
+    spi_set_csid(&spi_host, 0);
+
+    // Reset
+    const uint32_t reset_cmd = 0xFFFFFFFF; // ???
```

Comment:
@consanii , can we replace this command with the real reset please? as @ruben-roalvarez said it is wrong

Reply:
I can double-check the datasheet in any case, but maybe @consanii you can do three tests: write the FLASH without a reset, with all Fs reset, and with the reset you found in the datasheet (all without previously sending any command from the PC). Let's see if it fails for any of the cases

Reply:
I will do the test but I think it will work in both three cases, as the reset is not needed in the first place. Moreover, even if the reset was needed, it would be redundant as the reset command revert the board to the initial power-on state. And the board is already in the initial power on state before the write command (as it's just turned on).

Reply:
Amazing! Thanks!